src/client/pfs.go

```diff
@@ -276,15 +276,23 @@ func (c APIClient) FlushCommit(commits []*pfs.Commit, toRepos []*pfs.Repo) ([]*p
 // Blocks are content addressed and are thus identified by hashes of the content.
 // NOTE: this is lower level function that's used internally and might not be
 // useful to users.
-func (c APIClient) PutBlock(reader io.Reader) (*pfs.BlockRefs, error) {
-	putBlockClient, err := c.BlockAPIClient.PutBlock(context.Background())
+func (c APIClient) PutBlock(delimiter pfs.Delimiter, reader io.Reader) (blockRefs *pfs.BlockRefs, retErr error) {
+	writer, err := c.newPutBlockWriteCloser(delimiter)
 	if err != nil {
 		return nil, sanitizeErr(err)
 	}
-	if _, err := io.Copy(protostream.NewStreamingBytesWriter(putBlockClient), reader); err != nil {
-		return nil, err
-	}
-	return putBlockClient.CloseAndRecv()
+
+	defer func() {
+		err := writer.Close()
+		if err != nil && retErr == nil {
+			retErr = err
+		} else {
+			blockRefs = writer.blockRefs
```

Comment:
This seems to be a strange pattern.  So basically the function always return `blockRefs` as nil, but in the defer function, we set `blockRefs` to an actual value if `err == nil` or `retErr != nil`?  So if `err` is not nil and `retErr` is not nil, we still return blockRefs?